packages/backend-auth/package.json

```diff
@@ -20,6 +20,7 @@
   "license": "Apache-2.0",
   "dependencies": {
     "@aws-amplify/auth-construct": "^1.3.1",
+    "@aws-amplify/reference-auth-construct": "^0.1.0",
```

Comment:
Do we have to create `@aws-amplify/reference-auth-construct` ? I.e. is there a desire to use it in raw CDK apps?

It could be hosted internally in `backend-auth` instead of creating new package.

So far in our repo we have:
1. AuthConstruct - public because there was requirement to support raw CDK.
2. ConversationHandler - public because it needs to be visible on data repo side.
3. AmplifyFunction - private/internal, because there was no need to make it public
4. AmplifyStorage - private/internal, because there was no need to make it public

I suggest we make ReferenceAuthConstruct private initially. We can move it later to separate package if needed. Less public API is better.

Reply:
We could make it private; I don't think we have a use case for making it public unless customers want to take advantage of the custom resource logic

Reply:
Updated , reference-auth-construct is removed, code lives within auth-construct

Reply:
> code lives within auth-construct

This doesn't make it private, i.e. it's still exported somewhere. But definitely makes situation better.

To make this private we'd need to move the construct to `backend-auth` and not export it there. It's fine to have constructs in `backend-X` if they're supposed to be private (see functions, storage).

However, perhaps there's no harm in having this in `auth-construct` and saying that we support reference auth in raw CDK context as well. It's just that we sign up to support extra public API.

@Amplifiyer @josefaidt wdyt ?